### PR TITLE
Add acceptance test for checking changing sharing permissions

### PR DESCRIPTION
**Pull request by dpakach**
## Description

Add acceptance test for checking changing sharing permissions from 
1. read-write to upload-write-without-modify
2. upload-write-without-modify to read-write

## Related Issue

- Test for issue #35155

## How Has This Been Tested?

:robot: 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

File: tests/acceptance/features/bootstrap/WebUIFilesContext.php [hunk 1]
```diff
@@ -1596,21 +1596,27 @@ public function folderInputFieldTooltipTextShouldBeDisplayedOnTheWebUI(
...
 	 * @return void
 	 * @throws \Exception
 	 */
-	public function itShouldNotBePossibleToDeleteFileFolderUsingTheWebUI($name) {
+	public function itShouldNotBePossibleToDeleteFileFolderUsingTheWebUI($shouldOrNot, $name) {
+		$expect = $shouldOrNot !== "not";
+		$name = \trim($name, $name[0]);
 		try {
-			$this->deleteTheFileUsingTheWebUI($name, false);
+			$retries = $expect ? 1 : null;
```
Comment (skshetry) [inline tests/acceptance/features/bootstrap/WebUIFilesContext.php]: Is this used anywhere?

File: tests/acceptance/features/bootstrap/WebUIFilesContext.php [hunk 2]
```diff
@@ -1596,21 +1596,27 @@ public function folderInputFieldTooltipTextShouldBeDisplayedOnTheWebUI(
...
+	 * @Then /^it should (not|)\s?be possible to delete file ((?:'[^']*')|(?:"[^"]*")) using the webUI$/
 	 *
+	 * @param string $shouldOrNot
 	 * @param string $name
 	 *
 	 * @return void
 	 * @throws \Exception
 	 */
-	public function itShouldNotBePossibleToDeleteFileFolderUsingTheWebUI($name) {
+	public function itShouldNotBePossibleToDeleteFileFolderUsingTheWebUI($shouldOrNot, $name) {
```
Comment (skshetry) [inline tests/acceptance/features/bootstrap/WebUIFilesContext.php]: Can you split this function? It was too difficult for me to think about this.

File: tests/acceptance/features/bootstrap/WebUIFilesContext.php [hunk 3]
```diff
@@ -1602,13 +1602,61 @@ public function folderInputFieldTooltipTextShouldBeDisplayedOnTheWebUI(
...
+	 *
+	 * @param string $action
+	 * @param string $name
+	 * @param string $shouldOrNot
+	 *
+	 * @return void
+	 * @throws \Exception
+	 */
+	public function optionShouldNotBeAvailable($action, $name, $shouldOrNot) {
+		$possible = $shouldOrNot !== "not";
```
Comment (skshetry) [inline tests/acceptance/features/bootstrap/WebUIFilesContext.php]: 🤦🏼

File: tests/acceptance/features/lib/FilesPageBasic.php [hunk 4]
```diff
@@ -649,4 +650,18 @@ public function enableShowHiddenFilesSettings() {
...
+	/**
+	 * Return is New File/folder button is available in the webUI
+	 *
+	 * @return boolean
+	 */
+	public function isUploadButtonAvailable() {
+		$btn = $this->find("xpath", $this->newFileFolderButtonXpath);
+		if ($btn === null) {
+			return false;
+		} else {
```
Comment (skshetry) [inline tests/acceptance/features/lib/FilesPageBasic.php]: `else` not required here

File: tests/acceptance/features/lib/FilesPageElement/FileActionsMenu.php [hunk 5]
```diff
@@ -218,4 +218,25 @@ public function getRenameActionLabel() {
...
+	public function isActionLabelVisible($action) {
+		$xpathLocator = \sprintf($this->fileActionXpath, $action);
+		$this->waitTillElementIsNotNull($xpathLocator);
+		$button = $this->menuElement->find(
+			"xpath",
+			$xpathLocator
+		);
+		if (!$button) {
+			return false;
+		} else {
```
Comment (skshetry) [inline tests/acceptance/features/lib/FilesPageElement/FileActionsMenu.php]: `else` is not essential

File: tests/acceptance/features/webUISharingPublic/shareByPublicLink.feature [hunk 6]
```diff
@@ -407,4 +407,61 @@ Feature: Share by public link
     Then folder "newfolder1" should be listed on the webUI
     And folder "newfolder" should not be listed on the webUI
     #And folder "newfolder" should be listed on the webUI
-    And folder "test" should be listed on the webUI
\ No newline at end of file
+    And folder "test" should be listed on the webUI
+
+  Scenario: Permissions work correctly on public link share with upload-write-without-modify
```
Comment (paurakhsharma) [inline tests/acceptance/features/webUISharingPublic/shareByPublicLink.feature]: The scenario needs to have when step.

File: tests/acceptance/features/webUISharingPublic/shareByPublicLink.feature [hunk 7]
```diff
@@ -407,4 +407,61 @@ Feature: Share by public link
...
+    Given the user has created a new public link for folder "simple-folder" using the webUI with
+      | permission | upload-write-without-modify |
+    And the public accesses the last created public link using the webUI
+    Then the option to rename file "lorem.txt" should not be available in the webUI
+    And the option to delete file "lorem.txt" should not be available in the webUI
+    And the option to upload file should be available in the webUI
+
+  Scenario: Permissions work correctly on public link share with read-write
```
Comment (paurakhsharma) [inline tests/acceptance/features/webUISharingPublic/shareByPublicLink.feature]: The scenario needs to have when step

File: tests/acceptance/features/webUISharingPublic/shareByPublicLink.feature [hunk 8]
```diff
@@ -424,3 +424,60 @@ Feature: Share by public link
...
+  Scenario: User tries to upload existing file in public link share with permissions upload-write-without-modify
+    Given the user has created a new public link for folder "simple-folder" using the webUI with
+      | permission | upload-write-without-modify |
+    And the public accesses the last created public link using the webUI
+    When the user uploads file "lorem.txt" using the webUI
+    Then a notification should be displayed on the webUI with the text "The file lorem.txt already exists"
+    And file "lorem.txt" should be listed on the webUI
+    And file "lorem (2).txt" should not be listed on the webUI
+
+  Scenario: User tries to upload existing file in public link share with permissions upload-write-without-modify
```
Comment (individual-it) [inline tests/acceptance/features/webUISharingPublic/shareByPublicLink.feature]: `upload-write-without-modify` ? copy-paste problem?

**Review by individual-it — CHANGES_REQUESTED** [review]
looks generally good to me, just a small request

work for additional PRs:
- test with single files
- test through API

**Review by sharidas — APPROVED** [review]
LGTM :+1:

**Review by individual-it — APPROVED** [review]

**Comment by dpakach** [comment]
Backport on #35267

**Pull request merged**